includes/managers/class-fs-clone-manager.php

```diff
@@ -694,17 +733,96 @@ private function maybe_show_clone_admin_notice() {
             );
         }
 
+        /**
+         * Removes the notices from the storage if the context product is either no longer active on the context subsite or it's active but there's no longer any clone. This prevents the notices from being shown on the network-level admin page when they are no longer relevant.
+         *
+         * @author Leo Fajardo (@leorw)
+         * @since 2.5.1
+         *
+         * @return string[]
+         */
+        private function maybe_remove_notices() {
+            $notices = array(
+                'clone_resolution_options_notice' => $this->_notices->get_sticky( 'clone_resolution_options_notice', true ),
+                'temporary_duplicate_notice'      => $this->_notices->get_sticky( 'temporary_duplicate_notice', true ),
+            );
+
+            $instances = Freemius::_get_all_instances();
+
+            foreach ( $notices as $id => $notice ) {
+                if ( ! is_array( $notice ) ) {
+                    unset( $notices[ $id ] );
+                    continue;
+                }
+
+                if ( empty( $notice['data'] ) || ! is_array( $notice['data'] ) ) {
+                    continue;
+                }
+
+                if ( empty( $notice['data']['product_ids'] ) || empty( $notice['data']['blog_id'] ) ) {
+                    continue;
+                }
+
+                $product_ids = $notice['data']['product_ids'];
+                $blog_id     = $notice['data']['blog_id'];

```

Comment:
@fajardoleo it's very much possible that a subsite will be deleted and therefore, `switch_to_blog()` will not function as expected. Thus, I suggest using [`get_site()`](https://github.com/WordPress/wordpress-develop/blob/5.8.1/src/wp-includes/ms-site.php#L308) first to check if the result isn't `null` and only then continue with the logic.